server/server.go

```diff
@@ -404,136 +381,48 @@ func (s *server) waitChAndUpdateClusterInfo(msg string, readyCh chan struct{}, u
 // - network operation terminates successfully by setting CustomChainsHealthy
 func (s *server) WaitForHealthy(ctx context.Context, _ *rpcpb.WaitForHealthyRequest) (*rpcpb.WaitForHealthyResponse, error) {
 	s.log.Debug("WaitForHealthy")
-	if s.getNetwork() == nil {
-		return nil, ErrNotBootstrapped
-	}
-	ctx, cancel := context.WithTimeout(ctx, 2*time.Minute)
+
+	ctx, cancel := context.WithTimeout(ctx, waitForHealthyTimeout)
 	defer cancel()
-	var err error
-	continueLoop := true
-	for continueLoop {
+
+	for {
+		s.mu.RLock()
+
+		if s.network == nil {
+			s.mu.RUnlock()
+			return nil, ErrNetworkNotRunning
+		}
 		if s.clusterInfo.CustomChainsHealthy {
-			break
+			s.mu.RUnlock()
+			return &rpcpb.WaitForHealthyResponse{ClusterInfo: s.clusterInfo}, nil
 		}
+		s.mu.RUnlock()
+
 		select {
-		case err = <-s.asyncErrCh:
```

Comment:
we want to notify the cause of async error here (cause for the network to fail)

Reply:
I don't understand -- this channel doesn't exist anymore. Where are the unhandled errors?

Reply:
https://github.com/ava-labs/avalanche-network-runner/pull/447#discussion_r1131592992